Asterix/WSC_functions.py

```diff
@@ -84,56 +97,153 @@ def invertSVD(matrix_to_invert,
                 plt.show()
                 plt.close()
         pseudoinverse = np.dot(np.dot(np.transpose(V), InvS_truncated),
-                               np.transpose(U))
+                                np.transpose(U))
 
     return [np.diag(InvS), np.diag(InvS_truncated), pseudoinverse]
 
 
-def creatingInterractionmatrix(input_wavefront, testbed, dimEstim,
-                               amplitudeEFC, matrix_dir):
+def creatingInterractionmatrix(testbed: OptSy.Testbed,
+                               dimEstim,
+                               amplitudeEFC,
+                               matrix_dir,
+                               initial_DM_voltage=0.,
+                               input_wavefront=1.,
+                               MatrixType='',
+                               save_all_planes_to_fits=False,
+                               dir_save_all_planes=None,
+                               visu=False):
     """ --------------------------------------------------
-    Create the jacobian matrix for Electric Field Conjugation
+    Create the jacobian matrix for Electric Field Conjugation. The Matrix is not
+    limited to the DH size but to the whole FP [dimEstim, dimEstim]. 
+    First half is real part, second half is imag part.
+
+    The matrix size is therefore [total(DM.basis_size), 2*dimEstim^2]
+
+    We save the matrix in .fits independently for each DMs, only if the initial 
+    wavefront and DMs are flat.
+
+    This code works for all testbeds without prior assumption (if we have at 
+    least 1 DM of course). We have optimized the code to only do once optical 
```

Comment:
We have optimized ... Je ne comprends pas cette phrase.
"one" au lieu de "once" ?